components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/RequestFilter.java

```diff
@@ -32,31 +33,30 @@
  */
 public class RequestFilter {
 
-    public static Map<String, String[]> getFilteredHeaders(Map<String, String[]> headers, ActionType actionType) {
+    public static List<Header> getFilteredHeaders(List<Header> headers, ActionType actionType) {
 
-        Map<String, String[]> filteredHeaders = new HashMap<>();
+        List<Header> filteredHeaders = new ArrayList<>();
         Set<String> excludedHeaders = ActionExecutorConfig.getInstance()
                 .getExcludedHeadersInActionRequestForActionType(actionType);
 
-        headers.forEach((key, value) -> {
-            // Headers are case-insensitive. Therefore, convert the key to lowercase before comparing.
-            if (!excludedHeaders.contains(key.toLowerCase(Locale.ROOT))) {
-                filteredHeaders.put(key, value);
+        headers.forEach(header -> {
+            if (!excludedHeaders.contains(header.getName())) {
+                filteredHeaders.add(header);
```

Comment:
@Shenali-SJ I think my previous comment is a mistake. Ideally this lowecase check should still be there.

This is because the header list here actually includes the headers that came in the request to the server. And what this check do is to validate if those are excluded from the config. So since headers are case insensitive this check still apply and we should not change the headers came from request. So let's keep this check and remove the modification added in Header class. Tests need to update accordingly

Reply:
This is updated with b1aae9dc06fbd0613e3c1b079177f23ce977be61